### PR TITLE
rg-240: add dates to subscription table. Dependency: rg-138

### PR DESCRIPTION
**Pull request by ValentinaTenus**
1. added start/end date to subscription table
2. fixed issues with stripe env
3. fixed method for update subscription status
4. add subscription update status to event handlers

### CODE REVIEW

File: backend/src/bundles/payment-method/payment-method.model.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public 'card': string;
+
+    public 'expireDate': Date;
+
+    public static override get tableName(): typeof DatabaseTableName.PAYMENT_METHOD {
+        return DatabaseTableName.PAYMENT_METHOD;
+    }
+
+    public static getRelationMappings(): RelationMappings {
+        return {};
```
Comment (siefimov) [inline backend/src/bundles/payment-method/payment-method.model.ts:22]: Consider adding a relationship for the `userId` field in the `getRelationMappings()` method of the `PaymentMethodModel` class to establish a relationship with the `UserModel` class.

File: backend/src/bundles/payment/helpers/get-card-expire-date.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { type Stripe } from 'stripe';
+
+const getCardExpireDate = (card: Stripe.PaymentMethod.Card): Date => {
+    const { exp_month, exp_year } = card;
```
Comment (siefimov) [inline backend/src/bundles/payment/helpers/get-card-expire-date.ts:4]: Names of variables should be written in camelCase notation.

File: backend/src/bundles/payment/payment.service.ts [hunk 3]
```diff
@@ -119,18 +119,66 @@ class PaymentService implements IPaymentService {
         const subscription: Stripe.Subscription =
             await this.createStripeSubscription(customer.id, priceId);
 
-        const { latest_invoice, id } = subscription;
+        const { latest_invoice, id, status } = subscription;
```
Comment (siefimov) [inline backend/src/bundles/payment/payment.service.ts:122]: Names of variables should be written in camelCase notation.

**Review by siefimov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/payment/helpers/get-card-expire-date.ts (hunk 2 above)
Comment (ValentinaTenus) [reply backend/src/bundles/payment/helpers/get-card-expire-date.ts:4]: it is from stripe card

File: backend/src/bundles/payment/payment.service.ts (hunk 3 above)
Comment (ValentinaTenus) [reply backend/src/bundles/payment/payment.service.ts:122]: from stripe

**Review by siefimov — APPROVED** [review]

**Review by vlad-blip — APPROVED** [review]

**Comment by vlad-blip** [comment]
[add startDate/endDate to subscription table](https://trello.com/c/0OqdChsA/240-add-startdate-enddate-to-subscription-table)

File: backend/src/migrations/20240304112515_add_id_generation_to_database.ts
Comment (MaxPoplavskiy) [inline backend/src/migrations/20240304112515_add_id_generation_to_database.ts:1]: Don't edit migrations

**Review by MaxPoplavskiy — CHANGES_REQUESTED** [review]

File: backend/src/bundles/stripe-events/enums/stripe-events.enum.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 enum StripeEvents {
     SUBSCRIPTION_CREATED = 'customer.subscription.created',
+    SUBSCRIPTION_UPDATED = 'customer.subscription.updated',
```
[the remark this answers is not in the document]
Comment (ValentinaTenus) [reply backend/src/bundles/stripe-events/enums/stripe-events.enum.ts:3]: there was only for create subscription

File: backend/src/migrations/20240304112515_add_id_generation_to_database.ts
Comment (ValentinaTenus) [reply backend/src/migrations/20240304112515_add_id_generation_to_database.ts:1]: these changes from rg-180((

Comment (MaxPoplavskiy) [reply backend/src/migrations/20240304112515_add_id_generation_to_database.ts:1]: Try syncing to dev

**Review by MaxPoplavskiy — APPROVED** [review]

**Pull request closed without merging**